internal/replication/storage.go

```diff
@@ -0,0 +1,432 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package replication
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/IBM-Blockchain/bcdb-server/pkg/logger"
+	"github.com/pkg/errors"
+	"go.etcd.io/etcd/etcdserver/api/snap"
+	"go.etcd.io/etcd/pkg/fileutil"
+	"go.etcd.io/etcd/raft"
+	"go.etcd.io/etcd/raft/raftpb"
+	"go.etcd.io/etcd/wal"
+	"go.etcd.io/etcd/wal/walpb"
+)
+
+// MaxSnapshotFiles defines max number of etcd/raft snapshot files to retain
```

Comment:
storage layer is taken from `fabric/orderer/consensus/etcdraft/storage.go` Hence, I read the code but not doing a deep review.